lib/private/shell.ts

```diff
@@ -1,23 +1,23 @@
 import * as child_process from 'child_process';
+import { EventType, globalOutputHandler } from '../progress';
 
 export type Logger = (x: string) => void;
 
 export interface ShellOptions extends child_process.SpawnOptions {
   readonly quiet?: boolean;
-  readonly logger?: Logger;
   readonly input?: string;
 }
 
 /**
  * OS helpers
  *
- * Shell function which both prints to stdout and collects the output into a
- * string.
+ * Shell function which both emits the output to the configured output handler, and collects the output
+ * to return it as a string.
  */
 export async function shell(command: string[], options: ShellOptions = {}): Promise<string> {
-  if (options.logger) {
-    options.logger(renderCommandLine(command));
-  }
+  globalOutputHandler.publishEvent(EventType.START, command.join(' '));
```

Comment:
Why? This seems like a new log message which i'm not sure is correct (or belongs in this PR).